### PR TITLE
feat: SplitButton positioning improvements

### PR DESCRIPTION
**Pull request by jeremy8883**
* Change the menu positioning from `position: relative` to `position: absolute`. I noticed when opening the dropdown menu, the content below it was getting pushed down. I am surprised that this wasn't causing issues in perform-ui, but I fixed this issue anyway, as it will likely cause unexpected behaviour in the future. @vicchirino if you could just confirm that I haven't broken any behaviour from your end.

Before:

sb-before

After:

sb-after

* Fix how the dropdown menu would be positioned when we run out of room. There was some logic that would show the menu above the SplitButtons, if we didn't have enough room below. The problem was that sometimes we didn't even have room above, causing menu items to become unselectable. Example:

Screen Shot 2020-02-02 at 10 42 23 am

My first iteration at fixing the issue was to have the menu overlap the buttons when necessary. eg.

itteration1

However, I realised that this was covering the most important primary menu item (ie. the "Edit" button in that example), so that was no good. With the second iteration, now the menu still gets displayed above, but only if there's enough room. eg.

iteration2

[Previous related PR](https://github.com/cultureamp/kaizen-design-system/pull/120)

### CODE REVIEW

File: packages/component-library/draft/Kaizen/SplitButton/DropdownMenu.tsx [hunk 1]
```diff
@@ -1,25 +1,49 @@
...
-    bottom: number
-    left: number
-    right: number
-  } | null
+  buttonsBoundingRect: ClientRect | null // get by calling getBoundingClientRect()
   hideDropdownMenu: () => void
   dir?: Dir
 }
 
+export const _calculateMenuTop = (
```
[the remark this answers is not in the document]
Comment (sebpearce) [reply packages/component-library/draft/Kaizen/SplitButton/DropdownMenu.tsx]: Good question, I'm curious about this too. Might be worth asking in #pract_front_end_eng.

File: packages/component-library/draft/Kaizen/SplitButton/styles.scss [hunk 2]
```diff
@@ -125,6 +126,7 @@ $button-width: 124px;
 .default,
 .primary {
   @include ca-type-ideal-body-bold;
```
[the remark this answers is not in the document]
Comment (sebpearce) [reply packages/component-library/draft/Kaizen/SplitButton/styles.scss]: I think it's just that this particular case is negative – it's dynamically calculated. This is an artefact of our current baseline shift typography hack (it applies `position: relative` and a `top` rule to all type mixins (see pic). It's quite old and there's a wide range of mixed opinions about it. Design Systems has just started a 6-week cycle focused on typography, so we're in the middle of considering how we want to proceed with this very thing. Feedback is welcome if you have any strong opinions (#team_design_systems)!

image
(See https://github.com/cultureamp/kaizen-design-system/blob/master/packages/component-library/styles/type.scss)

File: tslint.json [hunk 3]
```diff
@@ -38,6 +38,10 @@
     },
     "member-access": {
       "severity": "none"
-    }
+    },
+    "variable-name": {
+      "options": ["allow-leading-underscore", "allow-pascal-case"]
+    },
+    "object-literal-key-quotes": [true, "as-needed"]
```
Comment (jeremy8883) [inline tslint.json]: Just confirming @ActuallyACat if these updates are alright (I noticed you added `tslint:latest` to the config). It allows for the `_` prefix convention for private methods (pending confirmation in #pract_front_end_eng if this is the way to go), and to only require `"` marks around object keys when needed.

Comment (ActuallyACat) [reply tslint.json]: Yup, I'm okay with this. 👍

**Pull request merged**